lib/rules/media-query-no-invalid/index.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+const {
+	isMediaQueryInvalid,
+	isGeneralEnclosed,
+	isMediaFeaturePlain,
+	isMediaFeatureRange,
+	isMediaFeatureBoolean,
+} = require('@csstools/media-query-list-parser');
+
+const atRuleParamIndex = require('../../utils/atRuleParamIndex');
+const parseMediaQuery = require('../../utils/parseMediaQuery');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const isCustomMediaQuery = require('../../utils/isCustomMediaQuery');
+const { rangeTypeMediaFeatureNames } = require('../../reference/mediaFeatures');
+
+const ruleName = 'media-query-no-invalid';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (query) => `Unexpected invalid media query "${query}"`,
+});
+
+const HAS_MIN_MAX_PREFIX = /^(?:min|max)-/i;
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/media-feature-no-invalid',
```

Comment:
[suggest] Please fix the typo?

```suggestion
	url: 'https://stylelint.io/user-guide/rules/media-query-no-invalid',
```

Note: I've opened #6965 to check such a typo.

Reply:
Thank you!